src/tests/test_logger.py

```diff
@@ -37,33 +39,44 @@ class TestLogger(unittest.TestCase):
 
     def test_fileConfig(self):
         """Put logging.dat with special pattern and check it was used"""
-        tmp = os.environ['BITMESSAGE_HOME'] = tempfile.gettempdir()
+        # FIXME: this doesn't always work, probably due to module
+        # load order. Workaround through .travis.yml
+        tmp = set_appdata_folder(tempfile.gettempdir())
         log_config = os.path.join(tmp, 'logging.dat')
         log_file = os.path.join(tmp, 'debug.log')
+        pass_ = 1
 
         def gen_log_config(pattern):
             """A small closure to generate logging.dat with custom pattern"""
             with open(log_config, 'wb') as dst:
                 dst.write(self.conf_template.format(log_file, pattern))
 
-        pattern = r' o_0 '
-        gen_log_config(pattern)
-
         try:
-            from pybitmessage.debug import logger, resetLogging
-            if not os.path.isfile(log_file):  # second pass
-                pattern = r' <===> '
-                gen_log_config(pattern)
-                resetLogging()
+            from pybitmessage.debug import logger, resetLogging, flush_logs
         except ImportError:
             self.fail('There is no package pybitmessage. Things gone wrong.')
-        finally:
-            os.remove(log_config)
+        if not os.path.isfile(log_file):  # second pass
+            pass_ = 2
+            pattern = r' <===> '
+        else:
+            pattern = r' o_0 '
+        gen_log_config(pattern)
+        resetLogging()
 
-        logger_ = logging.getLogger('default')
+        self.longMessage = True
 
-        self.assertEqual(logger, logger_)
+        logger1 = logging.getLogger('default')
 
-        logger_.info('Testing the logger...')
+        self.assertEqual(logger, logger1, ", pass {}".
+                         format(pass_))
 
-        self.assertRegexpMatches(open(log_file).read(), pattern)
+        logger2 = logging.getLogger('root')
```

Comment:
I don't understand the purpose, it's not used in code:
```shell
$ grep -r 'getLogger' src | grep root
src/tests/test_logger.py:        logger2 = logging.getLogger('root')
```

Reply:
I think it may be leftover from debugging, perhaps it isn't necessary anymore.